### PR TITLE
ENH: Improve UX of mutation threshold selectors and Warning: componentWillReceiveProps has been renamed

### LINKED ISSUE #460: ENH: Improve UX of mutation threshold selectors
I love the mutation threshold. Unfortunately, the UX is a bit cumbersome.

image

It would be cool if selections were remembered across different queries, say in a cookie or URL parameter or both

Another improvement would be if there were dropdowns to quickly select predermined values: 
e.g. 0.0%, 0.2%, 0.5%, 1%, 2%, 5%, 10%, 20%, 50%, 80%, 90%, 95%, 99% for the lower
e.g 0.1%, 0.5%, 2%, 5%, 10%, 20%, 50%, 80%, 90%, 95%, 99, 100% for the upper.

Typing in the current text boxes is very cumbersome due to the validation. So dropdown point and click would be cool.

### LINKED ISSUE #676: Warning: componentWillReceiveProps has been renamed
This warning is thrown by NumericInput in VariantMutation.tsx from the "react-numeric-input" library. The library seems to be unmaintained and will probably not work with React 18. 

Might be solved by using <input/> from mui, but the arrows/spinners only appear there, when hovering over the field.

Full error message:
`Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.

* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state

* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: NumericInput`

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves #460 #676

### CODE REVIEW

File: src/components/PercentageInput.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    { value: 20, label: '20%' },
+    { value: 40, label: '40%' },
+    { value: 60, label: '60%' },
+    { value: 80, label: '80%' },
+    { value: 100, label: '100%' },
+  ];
+
+  return (
+    <Slider
+      className={`${className}`}
```
Comment (fengelniederhammer) [inline src/components/PercentageInput.tsx]: This can probably be simplified?
```suggestion
      className={className}
```

**Review by chaoran-chen — APPROVED** [review]
Looks great! The slider and input field combination seems very useful to me. (@corneliusroemer, happy to hear your opinion :))

File: src/components/VariantMutations.tsx [hunk 2]
```diff
@@ -340,7 +327,7 @@ export const VariantMutations = ({ selector }: Props) => {
                   <Uniqueness value={aa.uniqueness} />)
                   <ul className='list-circle'>
                     {sortNucMutations(nucs, 'position')
-                      .filter(nuc => nuc.proportion >= minProportion && nuc.proportion <= maxProportion)
+                      .filter(nuc => nuc.proportion >= minProportion && nuc.proportion <= maxProportion / 100)
```
Comment (fengelniederhammer) [inline src/components/VariantMutations.tsx]: fix

File: src/components/VariantMutations.tsx [hunk 3]
```diff
@@ -329,7 +314,9 @@ export const VariantMutations = ({ selector }: Props) => {
 
           <MutationList className='list-disc' width={width ? width : 1}>
             {sortMergedEntries(data.mergedEntries, commonAAMutationsSort)
-              .filter(({ aa }) => aa.proportion >= minProportion && aa.proportion <= maxProportion)
+              .filter(
+                ({ aa }) => aa.proportion >= minProportion / 100 && aa.proportion <= maxProportion / 100
```
Comment (fengelniederhammer) [inline src/components/VariantMutations.tsx]: fix

File: src/components/PercentageInput.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { InputAdornment, OutlinedInput, Slider } from '@mui/material';
+
+export type PercentageInputProps = {
+  ratio: number;
+  setRatio: (percentage: number) => void;
+  className?: string;
+};
+
+export function PercentageInput({ ratio, setRatio, className }: PercentageInputProps) {
+  // workaround for js floating number problem
```
Comment (fengelniederhammer) [inline src/components/PercentageInput.tsx]: remove comment and put the code into a method with a descriptive name instead.

File: src/components/VariantMutations.tsx (hunk 2 above)
Comment (chaoran-chen) [reply src/components/VariantMutations.tsx]: Oh, interesting! In some parts of the code, there is `/ 100` and in other parts, there isn't. The proportions should be ideally stored as a float between 0 and 1, shouldn't it?

Comment (JonasKellerer) [reply src/components/VariantMutations.tsx]: Yeah. It should. :) It is a leftover from my own testing. I'll fix it in a bit.

**Pull request merged**